Core/Database/Repositories/EmpireRepository.cs

```diff
@@ -2,27 +2,67 @@
 using System.Data;
 using System.Threading.Tasks;
 using Dapper;
+using QuantumCore.Core.Cache;
 
 namespace QuantumCore.Database;
 
 public interface IEmpireRepository
 {
-    Task<byte?> GetEmpireForAccountAsync(Guid accountId);
+    Task<byte?> GetEmpireForPlayerAsync(Guid accountId);
+    Task<byte?> GetTempEmpireForAccountAsync(Guid accountId);
 }
 
 public class EmpireRepository : IEmpireRepository
 {
     private readonly IDbConnection _db;
+    private readonly ICacheManager _cacheManager;
 
-    public EmpireRepository(IDbConnection db)
+    public EmpireRepository(IDbConnection db, ICacheManager cacheManager)
     {
         _db = db;
+        _cacheManager = cacheManager;
     }
 
-    public async Task<byte?> GetEmpireForAccountAsync(Guid accountId)
+    public async Task<byte?> GetEmpireForPlayerAsync(Guid playerId)
     {
-        return await _db.QueryFirstOrDefaultAsync<byte>(
-            "SELECT Empire FROM account.accounts WHERE Id = @AccountId", new { AccountId = accountId });
+        var empireRedisKey = $"empire-pid:{playerId}";
+        var cachedEmpire = await _cacheManager.Get<byte?>(empireRedisKey);
+
+        if(cachedEmpire.HasValue)
+        {
+            // If found in Redis on Player, return the cached empire value
```

Comment:
Useless comment. It explains what the code does. The code is self explanatory here. Even if you don't understand C# you will get what this code does as it is well written.